src/Pact/Typechecker.hs

```diff
@@ -1204,7 +1205,10 @@ showFails = do
 
 -- | unsafe lens for using `typecheckBody` with const
 singLens :: Iso' a [a]
-singLens = iso pure head
+singLens = iso pure (\case
```

Comment:
I'd add `HasCallStack` constraint to this function to get better traces where it happened for debuggability (unless it's a hot spot, though I doubt it is).